scripts/hosts-add.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Add new IP-host pair to /etc/hosts.
+
+if [[ "$1" && "$2" ]]
+then
+    IP=$1
+    HOSTNAME=$2
+
+    if [ -n "$(grep $HOSTNAME /etc/hosts)" ]
```

Comment:
this is probably an edge case, and probably not worth holding up this merge to figure out, but would subdomains cause issues here? for example, if `admin.mysite.test` is already setup, and we try to add `mysite.test`, I believe your grep would find it.

Reply:
Same deal here; I didn't write this code, but yes, I believe you are correct. And while I agree that it shouldn't hold-up the merge, it would be trivial to fix. Good catch.

Reply:
I just fixed this by requiring that the match not begin with a `.`, which would prevent the scenario you describe.

You can see what I mean by examining the demo at https://regex101.com/r/V7lYpc/2 . For the expression to match, it would need to be changed from `[^.]homestead.site` to `[^.]docs.homestead.site`, which achieves what we want, if I'm not mistaken.

There's still a flaw with it, which is that the `.` character in `$HOSTNAME` is not matched literally, but rather, represents "any character". By no means a show-stopper, but worth mentioning.

The way to fix it would be to call the equivalent of https://php.net/preg_quote on `$HOSTNAME`, but I don't know enough Bash to do that without guessing.